base/Enums.jl

```diff
@@ -9,6 +9,9 @@ abstract Enum
 
 Base.convert{T<:Integer}(::Type{T}, x::Enum) = convert(T, box(Int32, x))
 
+Base.:(==)(x::Enum, y::Integer) = Int32(x) == y
+Base.:(==)(x::Integer, y::Enum) = x == Int32(y)
```

Comment:
I don't think these shouldnt exist. Enum and Integer aren't compatible types. Why was this needed?